### DISCUSSION TITLE
New Customizable Option Request: Automatically Save when Replace

### DISCUSSION THREAD

**Discussion by drcxd** [meow-edit/meow#346] [Ideas]
As the topic said, could we add a new custom option which allows `meow-replace` command automatically save the replaced content? This is similar to `s` in vim. I think people would like to have it?

**Comment by fbergroth** [comment]
There is an existing command `meow-replace-save` that seems to do this.

**Reply by drcxd** [reply]
I do not think so. What I want is that if I press C with "text" in the following sample selected, then "text" would be cut and saved, thus when I press P at somewhere else, "text" would be pasted. Using `mewo-replace-save` simply replace "text" with something I do not know where it comes from.

```
some random |text|
```

By the way, `meow-replace-save` is not documented.

**Reply by eshrh** [reply]
There might have been a misunderstanding. You're not looking for a `meow-replace` that saves the text, but a `meow-change` that saves the text.

You'll be interested in `meow-change-save` in this case. You're right about neither being documented though.

When you call `meow-replace-save`, the text under the cursor is *replaced* with what you have in your clipboard, and saved for future use.

**Reply by drcxd** [reply]
You are right. I intended to mean `meow-change-save`. Sorry for the confusion. Actually, I did test `meow-replace-save` before started this discussion and I thought it was `meow-change-save`. Anyway, thank you so much.

**Comment by DogLooksGood** [comment]
If you are looking for a wap to do quick text swapping, you can  do `grab`, then `grab-swap`.

**Reply by drcxd** [reply]
Thanks. I'll see if I can use this.
